pkg/splunk/enterprise/util.go

```diff
@@ -34,15 +37,244 @@ import (
 // kubernetes logger used by splunk.enterprise package
 var log = logf.Log.WithName("splunk.enterprise")
 
-// ApplyCommonSecretObject creates/updates the namespace scoped "splunk-secrets" K8S secret object
-func ApplyCommonSecretObject(client splcommon.ControllerClient, namespace string) (*corev1.Secret, error) {
+// GetVersionedSecretName returns a versioned secret name
+func GetVersionedSecretName(versionedSecretIdentifier string, version string) string {
+	return fmt.Sprintf("%s-secret-v%s", versionedSecretIdentifier, version)
```

Comment:
How about assigning "%s-secret-v%s" also a name in names.go ?

Reply:
I kept it as an API to avoid fmt.Sprintf's everywhere with the names.go approach.

Reply:
Other names such as `defaultsTemplateStr`, `serviceTemplateStr` follow the pattern which I mentioned, so I think its better to follow the same format which has been implemented already, just for the sake of consistency.

Reply:
Done